File: internal/forms/federal/schedule_3.go
package federal

import (
	"taxpilot/internal/forms"
)

// Schedule3 returns the FormDef for Schedule 3 — Additional Credits and Payments.
//
// Part I: Nonrefundable Credits (deferred — education, child, etc.)
// Part II: Other Payments and Refundable Credits
//   - Line 10: Estimated tax payments -> flows to 1040 line 26
//   - Line 15: Total other payments -> flows to 1040 line 31
func Schedule3() *forms.FormDef {
	return &forms.FormDef{
		ID:           forms.FormSchedule3,
		Name:         "Schedule 3 — Additional Credits and Payments",
		Jurisdiction: forms.Federal,
		TaxYears:     []int{2025},
		Fields: []forms.FieldDef{
			// --- Part I: Nonrefundable Credits ---

			// Line 1: Foreign tax credit (from Form 1116 line 22)
			{
				Line:      "1",
				Type:      forms.Computed,
				Label:     "Foreign tax credit",
				DependsOn: []string{"form_1116:22"},
				Compute: func(dv forms.DepValues) float64 {
					return dv.Get("form_1116:22")
				},
			},
			// Line 2: Child and dependent care credit (deferred)
			{
				Line:      "2",
				Type:      forms.Computed,
				Label:     "Child and dependent care credit",
				DependsOn: []string{},
				Compute: func(dv forms.DepValues) float64 {
					return 0
				},
			},
			// Line 3: Education credits (deferred)
			{
				Line:      "3",
				Type:      forms.Computed,
				Label:     "Education credits",
				DependsOn: []string{},
				Compute: func(dv forms.DepValues) float64 {
					return 0
				},
			},
			// Line 8: Total nonrefundable credits
			{
				Line:      "8",
				Type:      forms.Computed,
				Label:     "Total nonrefundable credits",
				DependsOn: []string{"schedule_3:1", "schedule_3:2", "schedule_3:3"},
				Compute: func(dv forms.DepValues) float64 {
					return dv.Get("schedule_3:1") +
						dv.Get("schedule_3:2") +
						dv.Get("schedule_3:3")
				},
			},

			// --- Part II: Other Payments and Refundable Credits ---

			// Line 10: Estimated tax payments (UserInput — taxpayer enters total paid)
			{
				Line:   "10",
				Type:   forms.UserInput,
				Label:  "Estimated tax payments for 2025",
				Prompt: "How much did you pay in federal estimated taxes for 2025?",
			},
			// Line 15: Total other payments and refundable credits
			{
				Line:      "15",
				Type:      forms.Computed,
				Label:     "Total other payments",
				DependsOn: []string{"schedule_3:10"},
				Compute: func(dv forms.DepValues) float64 {
					return dv.Get("schedule_3:10")
				},
			},
		},
	}
}

File: api/src/forms/federal/schedule_3.rs
use crate::domain::builders::*;
use crate::domain::field::*;
use crate::domain::form::*;

pub fn schedule_3() -> FormDef {
    FormDef {
        id: FORM_SCHEDULE_3.to_string(),
        name: "Schedule 3 — Additional Credits and Payments".to_string(),
        jurisdiction: Jurisdiction::Federal,
        tax_years: vec![2024, 2025, 2026],
        question_group: "deductions".to_string(),
        question_order: 6,
        fields: vec![
            // --- Part I: Nonrefundable Credits ---
            // Line 1: Foreign tax credit
            ref_field("1", "Foreign tax credit", F1116_LINE_22),
            // Line 2: Child and dependent care credit (deferred)
            zero_field("2", "Child and dependent care credit"),
            // Line 3: Education credits (deferred)
            zero_field("3", "Education credits"),
            // Line 8: Total nonrefundable credits
            sum_field("8", "Total nonrefundable credits", vec![SCHED_3_LINE_1, "schedule_3:2", "schedule_3:3"]),
            // --- Part II: Other Payments ---
            // Line 10: Estimated tax payments
            input_field("10", "Estimated tax payments for 2025", "How much did you pay in federal estimated taxes for 2025?"),
            // Line 15
            ref_field("15", "Total other payments", SCHED_3_LINE_10),
        ],
    }
}
